web/app/themes/clarity/inc/maintenance.php

```diff
@@ -0,0 +1,398 @@
+            add_action('admin_notices', function () use ($user_id, $notification_title, $notification_content, $notification_level, $notification_id) {
+                if (!get_user_meta($user_id, "maintenance_notification_{$notification_id}_dismissed")) {
+                    $this->show_notification_banner($notification_level, $notification_title, $notification_content);
+                }
+            });
+        }
+        // If the banner is dismissed, update the user_meta table, so it's not shown again.
+        $this->dismiss_notification_banner($notification_id);
+    }
+
+    /**
+     * Set up the maintenance submenu page
+     */
+    public function options_page(): void
+    {
+        add_submenu_page(
+            'options-general.php', /* Parent Menu Slug */
+            'Maintenance settings', /* Page Title */
+            'Maintenance', /* Menu Title */
+            $this->capability, /* Capability */
+            'maintenance', /* Menu Slug */
+            [$this, 'render_options_page'], /* Callback */
+        );
+    }
+
+    /**
+     * Renders the options page
+     */
+    public function render_options_page(): void
+    {
+
+        // check user capabilities
+        if (!current_user_can($this->capability)) {
+            return;
+        }
+
+        if (isset($_GET['settings-updated'])) {
+            add_settings_error('maintenance_messages', 'maintenance_message', __('Settings Saved', 'maintenance'), 'updated');
+        }
+
+        $options = get_option('maintenance_options');
+        $notification_title = $options['notification_title'] ?? '';
+        $notification_content = $options['notification_content'] ?? '';
+        $notification_level = $options['notification_level'] ?? 'info';
+
+        ?>
+        <div class="wrap">
+            <h1><?php echo esc_html(get_admin_page_title()); ?></h1>
+            <h2 class="description"></h2>
+            <form action="options.php" method="post">
+                <?php
+                settings_fields('maintenance');
+                do_settings_sections('maintenance');
+                ?>
+                <h3>Preview</h3>
+                <?php
+                // Display a preview of the notification banner
+                $this->show_notification_banner($notification_level, $notification_title, $notification_content, true);
+                submit_button('Save Settings');
+                ?>
+            </form>
+        </div>
+        <?php
+    }
+
+    /**
+     * Render a settings field.
+     *
+     * @param array $args A mixed array of arguments to render the different field types
+     */
+    public function render_field(array $args): void
+    {
+
+        $field = $args['field'];
+
+        // Get the value of the setting we've registered with register_setting()
+        $options = get_option('maintenance_options');
```

Comment:
Hey @EmilyHazlehurst - on the first test of the class I received a fatal error. This option array will only exist once values have been submitted. 

It is however possible to use the 2nd argument of the `get_option()` function to define a default array, for example:

```php
get_option('maintenance_options', [
    'notification_content' => '',
    'notification_title' => ''
]);
```

Reply:
https://developer.wordpress.org/reference/functions/get_option/

Reply:
```php
Warning: Trying to access array offset on value of type bool in /bedrock/web/app/themes/clarity/inc/maintenance.php on line 220
```

Reply:
Thanks @wilson1000, I've updated this to have default values.